### PR TITLE
[0.9.0::01] Make tests pass, fix imports and interfaces changes

### PR DESCRIPTION
**Pull request by MaksymilianDemitraszek**
This PR makes an initial effort to support cairo 0.9
- [x] Review cairo interfaces again
- [x] Unskip devnet tests

### CODE REVIEW

File: protostar/commands/deploy/starkware/starknet_cli.py [hunk 1]
```diff
@@ -30,25 +30,21 @@ async def deploy(
...
-    gateway_client = GatewayClient(
-        url=gateway_url, retry_config=RetryConfig(n_retries=1)
-    )
     if salt is not None and not salt.startswith("0x"):
         raise ValueError(f"salt must start with '0x'. Got: {salt}.")
 
-    numeric_salt: int = 0
     try:
-        numeric_salt = (
+        salt = (
```
Comment (kasperski95) [inline protostar/commands/deploy/starkware/starknet_cli.py:37]: The type of `salt` changes from `Optional[str]` to `int`.

**Review by kasperski95 — APPROVED** [review]
LGTM, but I need to take a closer look at this code after the official release of `cairo-lang` 0.9.0.

**Comment by kasperski95** [comment]
I think we should unskip devnet tests in a different PR.

File: protostar/commands/deploy/starkware/starknet_cli.py [hunk 2]
```diff
@@ -59,25 +58,23 @@ async def deploy(
...
+        url=gateway_url, retry_config=RetryConfig(n_retries=1)
+    )
     gateway_response = await gateway_client.add_transaction(tx=tx, token=token)
-
-    if gateway_response["code"] != StarkErrorCode.TRANSACTION_RECEIVED.name:
-        raise DeployContractException(
-            message=f"Failed to send transaction. Response: {gateway_response}"
-        )
-
-    # return gateway_response
```
Comment (kasperski95) [inline protostar/commands/deploy/starkware/starknet_cli.py:80]: I think we should leave the current code. This kind of errors should be handled by exceptions, not asserts.

**Review by kasperski95 — CHANGES_REQUESTED** [review]

File: protostar/commands/deploy/starkware/starknet_cli.py (hunk 2 above)
Comment (MaksymilianDemitraszek) [reply protostar/commands/deploy/starkware/starknet_cli.py:80]: Agreed

**Review by kasperski95 — APPROVED** [review]
LGTM

**Comment by kasperski95** [comment]
You can merge it to master

**Pull request merged**